### PR TITLE
Added find endpoint

### PR DESCRIPTION
**Pull request by Vasis3038**
https://trello.com/c/6EupE6wl/41-skillservice-add-findendpoint

### CODE REVIEW

File: src/SkillService.Models.Dto/Models/ShortSkillInfo.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace LT.DigitalOffice.SkillService.Models.Dto.Models
```
Comment (maybee16) [inline src/SkillService.Models.Dto/Models/ShortSkillInfo.cs]: Use file-scoped namespace

File: src/SkillService.Business/Commands/UserSkill/FindUserSkillCommand.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+using System.Threading.Tasks;
+using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.SkillService.Business.Commands.UserSkill.Interfaces;
+using LT.DigitalOffice.SkillService.Data.Interfaces;
+using LT.DigitalOffice.SkillService.Mappers.Models.Interfaces;
+using LT.DigitalOffice.SkillService.Models.Db;
+using LT.DigitalOffice.SkillService.Models.Dto.Models;
+using MassTransit.Initializers;
+
+namespace LT.DigitalOffice.SkillService.Business.Commands.UserSkill
```
Comment (maybee16) [inline src/SkillService.Business/Commands/UserSkill/FindUserSkillCommand.cs]: Use file-scoped namespace

File: src/SkillService.Business/Commands/UserSkill/Interfaces/IFindUserSkillCommand.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using LT.DigitalOffice.Kernel.Attributes;
+using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.SkillService.Models.Dto.Models;
+
+namespace LT.DigitalOffice.SkillService.Business.Commands.UserSkill.Interfaces
```
Comment (maybee16) [inline src/SkillService.Business/Commands/UserSkill/Interfaces/IFindUserSkillCommand.cs]: Use file-scoped namespace

File: src/SkillService.Data/UserSkillRepository.cs [hunk 4]
```diff
@@ -25,11 +25,12 @@ public async Task<List<Guid>> GetUserSkillIdsAsync(Guid userId)
         .Where(us => us.UserId == userId).Select(us => us.SkillId).ToListAsync();
     }
 
-    public async Task<List<DbUserSkill>> GetAsync(Guid userId)
+    public async Task<List<DbSkill>> FindAsync(Guid userId)
```
Comment (maybee16) [inline src/SkillService.Data/UserSkillRepository.cs]: Remove async await from here, you already use await in FindUserSkillCommand (line 31)

File: src/SkillService.Mappers/Models/ShortSkillInfoMapper.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+using LT.DigitalOffice.SkillService.Models.Db;
+using LT.DigitalOffice.SkillService.Models.Dto.Models;
+
+namespace LT.DigitalOffice.SkillService.Mappers.Models
+{
+  public class ShortSkillInfoMapper : IShortSkillInfoMapper
+  {
+    public ShortSkillInfo Map(DbSkill dbSkill)
+    {
+      if (dbSkill is null)
```
Comment (maybee16) [inline src/SkillService.Mappers/Models/ShortSkillInfoMapper.cs]: Maybe `return dbSkill is null ? null : new ShortSkillInfo{...};`

File: src/SkillService.Mappers/Models/Interfaces/IShortSkillInfoMapper.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+﻿using LT.DigitalOffice.Kernel.Attributes;
+using LT.DigitalOffice.SkillService.Models.Db;
+using LT.DigitalOffice.SkillService.Models.Dto.Models;
+
+namespace LT.DigitalOffice.SkillService.Mappers.Models.Interfaces
```
Comment (maybee16) [inline src/SkillService.Mappers/Models/Interfaces/IShortSkillInfoMapper.cs]: Use file-scoped namespace

File: src/SkillService.Mappers/Models/ShortSkillInfoMapper.cs [hunk 7]
```diff
@@ -0,0 +1,23 @@
+﻿using LT.DigitalOffice.SkillService.Mappers.Models.Interfaces;
+using LT.DigitalOffice.SkillService.Models.Db;
+using LT.DigitalOffice.SkillService.Models.Dto.Models;
+
+namespace LT.DigitalOffice.SkillService.Mappers.Models
```
Comment (maybee16) [inline src/SkillService.Mappers/Models/ShortSkillInfoMapper.cs]: Use file-scoped namespace

File: src/SkillService.Models.Dto/Models/ShortSkillInfo.cs (hunk 1 above)
Comment (Vasis3038) [reply src/SkillService.Models.Dto/Models/ShortSkillInfo.cs]: Done

File: src/SkillService.Business/Commands/UserSkill/FindUserSkillCommand.cs (hunk 2 above)
Comment (Vasis3038) [reply src/SkillService.Business/Commands/UserSkill/FindUserSkillCommand.cs]: Done

File: src/SkillService.Data/UserSkillRepository.cs (hunk 4 above)
Comment (Vasis3038) [reply src/SkillService.Data/UserSkillRepository.cs]: Done

File: src/SkillService.Mappers/Models/Interfaces/IShortSkillInfoMapper.cs (hunk 6 above)
Comment (Vasis3038) [reply src/SkillService.Mappers/Models/Interfaces/IShortSkillInfoMapper.cs]: Done

File: src/SkillService.Mappers/Models/ShortSkillInfoMapper.cs (hunk 7 above)
Comment (Vasis3038) [reply src/SkillService.Mappers/Models/ShortSkillInfoMapper.cs]: Done

File: src/SkillService.Mappers/Models/ShortSkillInfoMapper.cs (hunk 5 above)
Comment (Vasis3038) [reply src/SkillService.Mappers/Models/ShortSkillInfoMapper.cs]: Yep

File: src/SkillService.Business/Commands/UserSkill/Interfaces/IFindUserSkillCommand.cs (hunk 3 above)
Comment (Vasis3038) [reply src/SkillService.Business/Commands/UserSkill/Interfaces/IFindUserSkillCommand.cs]: Done

File: src/SkillService.Business/Commands/UserSkill/Interfaces/IFindUserSkillCommand.cs [hunk 8]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using LT.DigitalOffice.Kernel.Attributes;
+using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.SkillService.Models.Dto.Models;
+
+namespace LT.DigitalOffice.SkillService.Business.Commands.UserSkill.Interfaces;
+[AutoInject]
```
Comment (Levliv) [inline src/SkillService.Business/Commands/UserSkill/Interfaces/IFindUserSkillCommand.cs:10]: blank line

File: src/SkillService.Business/Commands/UserSkill/FindUserSkillCommand.cs [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+using LT.DigitalOffice.Kernel.Responses;
+using LT.DigitalOffice.SkillService.Business.Commands.UserSkill.Interfaces;
+using LT.DigitalOffice.SkillService.Data.Interfaces;
+using LT.DigitalOffice.SkillService.Mappers.Models.Interfaces;
+using LT.DigitalOffice.SkillService.Models.Db;
+using LT.DigitalOffice.SkillService.Models.Dto.Models;
+using MassTransit.Initializers;
+
+namespace LT.DigitalOffice.SkillService.Business.Commands.UserSkill;
+public class FindUserSkillCommand : IFindUserSkillCommand
```
Comment (Levliv) [inline src/SkillService.Business/Commands/UserSkill/FindUserSkillCommand.cs:15]: blank line

File: src/SkillService.Mappers/Models/Interfaces/IShortSkillInfoMapper.cs [hunk 10]
```diff
@@ -0,0 +1,10 @@
+﻿using LT.DigitalOffice.Kernel.Attributes;
+using LT.DigitalOffice.SkillService.Models.Db;
+using LT.DigitalOffice.SkillService.Models.Dto.Models;
+
+namespace LT.DigitalOffice.SkillService.Mappers.Models.Interfaces;
+[AutoInject]
```
Comment (Levliv) [inline src/SkillService.Mappers/Models/Interfaces/IShortSkillInfoMapper.cs:7]: blank line

File: src/SkillService.Mappers/Models/ShortSkillInfoMapper.cs [hunk 11]
```diff
@@ -0,0 +1,18 @@
+﻿using LT.DigitalOffice.SkillService.Mappers.Models.Interfaces;
+using LT.DigitalOffice.SkillService.Models.Db;
+using LT.DigitalOffice.SkillService.Models.Dto.Models;
+
+namespace LT.DigitalOffice.SkillService.Mappers.Models;
+public class ShortSkillInfoMapper : IShortSkillInfoMapper
```
Comment (Levliv) [inline src/SkillService.Mappers/Models/ShortSkillInfoMapper.cs:7]: blank line

File: src/SkillService.Models.Dto/Models/ShortSkillInfo.cs [hunk 12]
```diff
@@ -0,0 +1,8 @@
+﻿using System;
+
+namespace LT.DigitalOffice.SkillService.Models.Dto.Models;
+public record ShortSkillInfo
```
Comment (Levliv) [inline src/SkillService.Models.Dto/Models/ShortSkillInfo.cs:5]: blank line

**Comment by Levliv** [comment]
BTW, is it Ok, that there are no configurations for DB model at all?

File: src/SkillService.Business/Commands/UserSkill/FindUserSkillCommand.cs [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+  {
+    _userSkillRepository = userSkillRepository;
+    _shotrSkillInfoMapper = shotrSkillInfoMapper;
+  }
+
+  public async Task<OperationResultResponse<List<ShortSkillInfo>>> ExecuteAsync(Guid userId)
+  {
+    OperationResultResponse<List<ShortSkillInfo>> response = new();
+    List<DbSkill> dbSkillsList = await _userSkillRepository.FindAsync(userId);
+    response.Body = dbSkillsList.Select(_shotrSkillInfoMapper.Map).ToList();
```
Comment (vorobevamv) [inline src/SkillService.Business/Commands/UserSkill/FindUserSkillCommand.cs]: why not (29-31)
OperationResultResponse<List'<ShortSkillInfo'>'> response = new(
body : (await _userSkillRepository.FindAsync(userId)).Select(_shotrSkillInfoMapper.Map).ToList());
?

**Review by vorobevamv — APPROVED** [review]

**Review by AlfaRedDog — APPROVED** [review]

File: src/SkillService.Business/Commands/UserSkill/Interfaces/IFindUserSkillCommand.cs (hunk 8 above)
Comment (Vasis3038) [reply src/SkillService.Business/Commands/UserSkill/Interfaces/IFindUserSkillCommand.cs:10]: Done

File: src/SkillService.Business/Commands/UserSkill/FindUserSkillCommand.cs (hunk 9 above)
Comment (Vasis3038) [reply src/SkillService.Business/Commands/UserSkill/FindUserSkillCommand.cs:15]: Done

File: src/SkillService.Mappers/Models/Interfaces/IShortSkillInfoMapper.cs (hunk 10 above)
Comment (Vasis3038) [reply src/SkillService.Mappers/Models/Interfaces/IShortSkillInfoMapper.cs:7]: Done

File: src/SkillService.Models.Dto/Models/ShortSkillInfo.cs (hunk 12 above)
Comment (Vasis3038) [reply src/SkillService.Models.Dto/Models/ShortSkillInfo.cs:5]: Done

File: src/SkillService.Business/Commands/UserSkill/FindUserSkillCommand.cs (hunk 13 above)
Comment (Vasis3038) [reply src/SkillService.Business/Commands/UserSkill/FindUserSkillCommand.cs]: Done

File: src/SkillService.Mappers/Models/ShortSkillInfoMapper.cs (hunk 11 above)
Comment (Vasis3038) [reply src/SkillService.Mappers/Models/ShortSkillInfoMapper.cs:7]: Done

**Review by Levliv — APPROVED** [review]

**Review by maybee16 — APPROVED** [review]

**Review by VLiamin — APPROVED** [review]

**Pull request merged**